### PR TITLE
Refactored CircleCI workflows

### PR DESCRIPTION
**Pull request by ajsantander**
Refactors CI to follow these rules:

1. PRs to main branches (master, staging, develop) will require their corresponding workflows to run and pass (workflow-develop, workflow-staging, workflow)
2. To trigger each workflow, the name of the target / base branch needs to be included in the name of the child branch.
3. TL;DR: Always name new branches with "base_branch-new_branch"

Nevermind "master" in the branch name. It's just a trick to trigger ALL workflows in this PR.

### CODE REVIEW

File: .circleci/src/workflows/workflow-master.yml [hunk 1]
```diff
@@ -0,0 +1,31 @@
+jobs:
+  - job-prepare
+  # ~~~~~~~~~~~~ Deployment validation on MAINNET ~~~~~~~~~~~~~
+  - job-validate-deployments:
+      name: job-validate-deployments-mainnet
```
Comment (justinjmoses) [inline .circleci/src/workflows/workflow-master.yml:5]: Did we not want to make this staging?

File: .circleci/src/workflows/workflow-staging.yml [hunk 2]
```diff
@@ -6,6 +6,36 @@ jobs:
           only: /.*(staging|master).*/
       requires:
         - job-prepare
+  # ~~~~~~~~~~~~ Deployment validation on MAINNET ~~~~~~~~~~~~~
+  - job-validate-deployments:
+      name: job-validate-deployments-mainnet
+      filters:
+        branches:
+          only: /.*(master).*/
```
Comment (justinjmoses) [inline .circleci/src/workflows/workflow-staging.yml:14]: shouldn't this be staging or master?

Comment (ajsantander) [reply .circleci/src/workflows/workflow-staging.yml:14]: Ah! Good catch 🙏

Comment (ajsantander) [reply .circleci/src/workflows/workflow-staging.yml]: Fixed

File: .circleci/src/workflows/workflow-master.yml (hunk 1 above)
Comment (ajsantander) [reply .circleci/src/workflows/workflow-master.yml]: I split this job in two, and moved one to staging:

job-validate-deployments | Validates deployment json data against on-chain data | workflow-staging
job-validate-etherscan | Validates that sources have been verified in etherscan | workflow-master

**Review by justinjmoses — APPROVED** [review]

**Review by justinjmoses — APPROVED** [review]

**Pull request merged**